conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -166,26 +166,20 @@ def generate(self, env=None, scope="build"):
         self.generate_args()
         VCVars(self._conanfile).generate(scope=scope)
 
+    def _shared_static_args(self):
+        args = []
+        if self._conanfile.options.get_safe("shared", False):
+            args.extend(["--enable-shared", "--disable-static"])
+        else:
+            args.extend(["--disable-shared", "--enable-static", "--with-pic"
+                        if self._conanfile.options.get_safe("fPIC", True)
+                        else "--without-pic"])
+        return args
+
     def generate_args(self):
         configure_args = []
+        configure_args.extend(self._shared_static_args())
```

Comment:
Unconditionally adding these prevents using the `--enable-option-checking=fatal` option with projects that do not support these flags. For example, `make` and `m4` do not understand these flags.